boards/arm/mimx8mm_evk/doc/index.rst

```diff
@@ -0,0 +1,182 @@
+
+The MIMX8MM EVK board doesn't have QSPI flash for the M4 and it needs
+to be started by the A53 core. The A53 core is responsible to load the M4 binary
+application into the RAM, put the M4 in reset, set the M4 Program Counter and
+Stack Pointer, and get the M4 out of reset. The A53 can perform these steps at
+bootloader level or after the Linux system has booted.
+
+The M4 can use up to 3 different RAMs. These are the memory mapping for A53 and M4:
+
++------------+-------------------------+------------------------+-----------------------+----------------------+
+| Region     | Cortex-A53              | Cortex-M4 (System Bus) | Cortex-M4 (Code Bus)  | Size                 |
++============+=========================+========================+=======================+======================+
+| OCRAM      | 0x00900000-0x0093FFFF   | 0x20200000-0x2023FFFF  | 0x00900000-0x0093FFFF | 256KB                |
++------------+-------------------------+------------------------+-----------------------+----------------------+
+| TCMU       | 0x00800000-0x0081FFFF   | 0x20000000-0x2001FFFF  |                       | 128KB                |
++------------+-------------------------+------------------------+-----------------------+----------------------+
+| TCML       | 0x007E0000-0x007FFFFF   |                        | 0x1FFE0000-0x1FFFFFFF | 128KB                |
++------------+-------------------------+------------------------+-----------------------+----------------------+
+| OCRAM_S    | 0x00180000-0x00187FFF   | 0x20180000-0x20187FFF  | 0x00180000-0x00187FFF | 32KB                 |
++------------+-------------------------+------------------------+-----------------------+----------------------+
+
+For more information about memory mapping see the
+`i.MX 8M Applications Processor Reference Manual`_  (section 2.1.2 and 2.1.3)
+
+At compilation time you have to choose which RAM will be used. This
+configuration is done in the file ``boards/arm/mimx8mm_evk/mimx8mm_evk.dts``
+with "zephyr,flash" (when CONFIG_XIP=y) and "zephyr,sram" properties.
+The available configurations are:
+
+.. code-block:: none
+
+   "zephyr,flash"
+   - &tcml_code
+   - &ocram_code
+   - &ocram_s_code
+
+   "zephyr,sram"
+   - &tcmu_sys
+   - &ocram_sys
+   - &ocram_s_sys
+
+Load and run Zephyr on M4 from A53 using u-boot by copying the compiled
+``zephyr.bin`` to the first FAT partition of the SD card and plug the SD
+card into the board. Power it up and stop the u-boot execution at prompt.
+
+Load the M4 binary onto the desired memory and start its execution using:
+
+.. code-block:: console
+
+   fatload mmc 0:1 0x7e0000 zephyr.bin;bootaux 0x7e0000
+
+Debugging
+=========
+
+MIMX8MM EVK board can be debugged by connecting an external JLink
+JTAG debugger to the J902 debug connector. Then download and install
+`J-Link GDB Server`_.
+
+Plug in the J-Link into the board and PC and run the J-Link command line tool:
+
+.. code-block:: console
+
+   JLinkGDBServer -device MIMX8MD6_M4
```

Comment:
instead of manually starting the JLinkGDBServer, can you use `west debug` or `west attach`? you'll need to add a `board.cmake` file (see other imx rt or kinetis boards for an example)

Reply:
Okay.